src/Protocol/ProtocolRecognizer.h

```diff
@@ -40,130 +37,67 @@ class cProtocolRecognizer:
-	virtual void SendSetSubTitle                (const cCompositeChat & a_SubTitle) override;
-	virtual void SendSetRawSubTitle             (const AString & a_SubTitle) override;
-	virtual void SendSetTitle                   (const cCompositeChat & a_Title) override;
-	virtual void SendSetRawTitle                (const AString & a_Title) override;
-	virtual void SendSoundEffect                (const AString & a_SoundName, double a_X, double a_Y, double a_Z, float a_Volume, float a_Pitch) override;
-	virtual void SendSoundParticleEffect        (const EffectID a_EffectID, int a_SrcX, int a_SrcY, int a_SrcZ, int a_Data) override;
-	virtual void SendSpawnEntity                (const cEntity & a_Entity) override;
-	virtual void SendSpawnMob                   (const cMonster & a_Mob) override;
-	virtual void SendStatistics                 (const cStatManager & a_Manager) override;
-	virtual void SendTabCompletionResults       (const AStringVector & a_Results) override;
-	virtual void SendThunderbolt                (int a_BlockX, int a_BlockY, int a_BlockZ) override;
-	virtual void SendTitleTimes                 (int a_FadeInTicks, int a_DisplayTicks, int a_FadeOutTicks) override;
-	virtual void SendTimeUpdate                 (Int64 a_WorldAge, Int64 a_TimeOfDay, bool a_DoDaylightCycle) override;
-	virtual void SendUnleashEntity              (const cEntity & a_Entity) override;
-	virtual void SendUnloadChunk                (int a_ChunkX, int a_ChunkZ) override;
-	virtual void SendUpdateBlockEntity          (cBlockEntity & a_BlockEntity) override;
-	virtual void SendUpdateSign                 (int a_BlockX, int a_BlockY, int a_BlockZ, const AString & a_Line1, const AString & a_Line2, const AString & a_Line3, const AString & a_Line4) override;
-	virtual void SendUseBed                     (const cEntity & a_Entity, int a_BlockX, int a_BlockY, int a_BlockZ) override;
-	virtual void SendWeather                    (eWeather a_Weather) override;
-	virtual void SendWholeInventory             (const cWindow & a_Window) override;
-	virtual void SendWindowClose                (const cWindow & a_Window) override;
-	virtual void SendWindowOpen                 (const cWindow & a_Window) override;
-	virtual void SendWindowProperty             (const cWindow & a_Window, short a_Property, short a_Value) override;
-
-	virtual AString GetAuthServerID(void) override;
-
-	virtual void SendData(const char * a_Data, size_t a_Size) override;
-
-protected:
-
-	/** The recognized protocol */
-	std::unique_ptr<cProtocol> m_Protocol;
+	/** Returns if we contain a concrete protocol corresponding to the client's protocol version. */
+	bool VersionRecognitionSuccessful()
+	{
+		return m_Protocol != nullptr;
+	}
 
-	/** Buffer for the incoming data until we recognize the protocol */
-	cByteBuffer m_Buffer;
+	/** Convenience overload to enable redirecting sends to the underlying implementation. */
+	auto & operator->()
+	{
+		return m_Protocol;
```

Comment:
Instead of pretending to be a pointer, why not just have a `GetConcreteProtocol` or something?

Reply:
We unfortunately have a lot of m_Protocol->SendSaucyXXX inside cClientHandle, so this avoids a mass-replace